client/packages/programs/src/JsonForms/components/Prescription/Prescription.tsx

```diff
@@ -56,6 +71,14 @@ const UIComponent = (props: ControlProps) => {
 
   const { success } = useNotification();
 
+  const itemCategoryPath = uischema.options?.['itemCategoryPath'];
```

Comment:
We'd normally do `const itemCategoryPath = options.itemCategoryPath;`

But I noticed that the `options` object returned by the Zod checker is `undefined` on first render. This was making the logic to detect a change of category far too complicated, until I realised I could pull the value directly  from `uischema.options`. It's the same value, and Zod is still doing run-time type-checking.

Reply:
nice find!